File: src/lib/openwig/ZObject.ts
import { EventTable } from './EventTable';

export class ZObject extends EventTable {
  public id: string;
  public name: string = "";
  public description: string = "";
  public visible: boolean = true;
  public active: boolean = true;
  public media: any = null;

  constructor(id?: string) {
    super();
    this.id = id || Math.random().toString(36).substring(7);
    this.table['Id'] = this.id;
    this.table['Name'] = this.name;
    this.table['Description'] = this.description;
    this.table['Visible'] = this.visible;
    this.table['Active'] = this.active;
  }

  public updateFromTable() {
    this.name = this.table['Name'] || this.name;
    this.description = this.table['Description'] || this.description;
    this.visible = this.table['Visible'] !== false;
    this.active = this.table['Active'] !== false;
  }
}

File: java-backport/net/jakedot/jourwigo/openwig/ZObject.java
package net.jakedot.jourwigo.openwig;

import java.util.UUID;

public class ZObject extends EventTable {
    public String id;
    public String name = "";
    public String description = "";
    public boolean visible = true;
    public boolean active = true;
    public Object media = null;

    public ZObject() {
        this(UUID.randomUUID().toString());
    }

    public ZObject(String id) {
        this.id = id;
        table.put("Id", this.id);
        table.put("Name", this.name);
        table.put("Description", this.description);
        table.put("Visible", this.visible);
        table.put("Active", this.active);
    }

    public void updateFromTable() {
        if (table.containsKey("Name")) this.name = (String) table.get("Name");
        if (table.containsKey("Description")) this.description = (String) table.get("Description");
        if (table.containsKey("Visible")) this.visible = (Boolean) table.get("Visible");
        if (table.containsKey("Active")) this.active = (Boolean) table.get("Active");
    }
}
